pkg/occlient/occlient_test.go

```diff
@@ -406,3 +408,883 @@ func Test_parseImageName(t *testing.T) {
+					t.Errorf("wrong number of VolumeMounts deleted, expected: %s, removed %d", "1 volumeMount", lenInitContainer-len(updatedDc.Spec.Template.Spec.Containers[0].VolumeMounts))
+				}
+
+				for _, initContainer := range updatedDc.Spec.Template.Spec.InitContainers {
+					if initContainer.Name == "copy-files-to-volume" {
+						t.Errorf("could not remove 'copy-volume-to-volume' InitContainer and instead some other InitContainer was removed")
+					}
+				}
+
+				for _, volume := range updatedDc.Spec.Template.Spec.Volumes {
+					if volume.Name == getAppRootVolumeName(tt.dcName) {
+						t.Errorf("could not remove %s volume", getAppRootVolumeName(tt.dcName))
+					}
+				}
+
+				for _, volumeMount := range updatedDc.Spec.Template.Spec.Containers[0].VolumeMounts {
+					if volumeMount.Name == getAppRootVolumeName(tt.dcName) {
+						t.Errorf("could not remove %s volumeMount", getAppRootVolumeName(tt.dcName))
+					}
+				}
+			} else if err == nil && tt.wantErr {
+				t.Errorf("test failed, expected: %s, got %s", "error", "no error")
+			} else if err != nil && !tt.wantErr {
+				t.Errorf("test failed, expected: %s, got %s", "no error", "error: "+err.Error())
+			}
+		})
+	}
+}
+
+func TestClient_GetBuildConfig(t *testing.T) {
+	tests := []struct {
+		name                string
+		buildName           string
+		projectName         string
+		returnedBuildConfig buildv1.BuildConfig
+		wantErr             bool
+	}{
+		{
+			name:        "buildConfig is found without error",
+			buildName:   "nodejs",
+			projectName: "project-app",
+			returnedBuildConfig: buildv1.BuildConfig{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "nodejs",
+				},
+			},
+			wantErr: false,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			fkclient, fkclientset := FakeNew()
+
+			fkclientset.BuildClientset.PrependReactor("get", "buildconfigs", func(action ktesting.Action) (handled bool, ret runtime.Object, err error) {
+				buildName := action.(ktesting.GetAction).GetName()
+				if buildName != tt.buildName {
+					return true, nil, fmt.Errorf("'get' was called with wrong buildName")
+				}
+				return true, &tt.returnedBuildConfig, nil
+			})
+
+			build, err := fkclient.GetBuildConfig(tt.buildName, tt.projectName)
+			if err == nil && !tt.wantErr {
+				// Check for validating actions performed
+				if (len(fkclientset.BuildClientset.Actions()) != 1) && (tt.wantErr != true) {
+					t.Errorf("expected 1 action in GetBuildConfig got: %v", fkclientset.AppsClientset.Actions())
+				}
+				if build.Name != tt.buildName {
+					t.Errorf("wrong GetBuildConfig got: %v, expected: %v", build.Name, tt.buildName)
+				}
+			} else if err == nil && tt.wantErr {
+				t.Errorf("test failed, expected: %s, got %s", "error", "no error")
+			} else if err != nil && !tt.wantErr {
+				t.Errorf("test failed, expected: %s, got %s", "no error", "error: "+err.Error())
+			}
+		})
+	}
+}
+
+func TestClient_UpdateBuildConfig(t *testing.T) {
```

Comment:
Oops, all `buildconfig` functions were already part of my PR #478

Reply:
@ashetty1 Should I remove them?

Reply:
For starters, can you review my PR #478 , at least the overlapping buildconfigs part, and see if you think I need to change anything?